### PR TITLE
helm: update updateStrategy 

### LINKED ISSUE #2335: Pod anti-affinity breaks upgrades
I have tired to upgrade ceph chars from 3.3.0 to 3.4.0 and ended up in a inconsistent state, with a new 3.4.0 pod in the pending state due to https://github.com/ceph/ceph-csi/blob/d3beaeb014fbc3a7f605054ab03006d127ac0430/charts/ceph-csi-rbd/templates/provisioner-deployment.yaml#L28-L42

In my case I have 3 workers and 3 replicas of ceph-provisioners, this leaves no node to start a new pod, as anti-affinity rules above prevents pod starting.
`
Selector:               app=ceph-csi-rbd,component=provisioner,release=ceph-csi-rbd
Replicas:               3 desired | 1 updated | 4 total | 3 available | 1 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
`

### PR DESCRIPTION
**Pull request by Yuggupta27**
Update the provisioner and nodeplugin updatestrategy
to allow maxUnavailable and maxSurge
pods at a time to be 50%

Fixes: #2335 

Signed-off-by: Yug Gupta <yuggupta27@gmail.com>

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

**Comment by Yuggupta27** [comment]
Both jobs failed with #2264

File: charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml
Comment (nixpanic) [inline charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml:17]: this should only be included when `Values.provisioner.strategy.type == RollingUpdate`

File: charts/ceph-csi-cephfs/values.yaml
Comment (nixpanic) [inline charts/ceph-csi-cephfs/values.yaml]: why delete? would it not help to change the values? if you delete this, should `updateStrategy` not be changed too?

File: charts/ceph-csi-cephfs/templates/nodeplugin-daemonset.yaml
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/templates/nodeplugin-daemonset.yaml]: Do we need it for daemonset pod also?

File: charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml
Comment (Madhu-1) [reply charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml:17]: Yes, need an if check here to check the `strategy.type == RollingUpdate`

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
The issue exists only for deployment, not for daemonset pods? can you please send a separate PR daemonset if it's required?

File: charts/ceph-csi-cephfs/templates/nodeplugin-daemonset.yaml
Comment (Yuggupta27) [reply charts/ceph-csi-cephfs/templates/nodeplugin-daemonset.yaml]: Sure, not required for now. Updated PR.

File: charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml
Comment (Yuggupta27) [reply charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml:17]: Thanks for the suggestion! Added the check.

**Review by Madhu-1 — APPROVED** [review]

**Comment by Yuggupta27** [comment]
`ci/centos/upgrade-tests-rbd` failed with 
```  Aug  4 11:34:39.008: failed to create vault statefulset failed to run kubectl: error running /usr/bin/kubectl --server=https://192.168.39.142:8443 --kubeconfig=/root/.kube/config --namespace=cephcsi-e2e-1f634079 create -f -:

    Command stdout:

    

    stderr:

    Error from server (AlreadyExists): error when creating "STDIN": serviceaccounts "rbd-csi-vault-token-review" already exists

    Error from server (AlreadyExists): error when creating "STDIN": clusterroles.rbac.authorization.k8s.io "rbd-csi-vault-token-review" already exists

    Error from server (AlreadyExists): error when creating "STDIN": clusterrolebindings.rbac.authorization.k8s.io "rbd-csi-vault-token-review" already exists```

**Comment by humblec** [comment]
LGTM. However will wait for review from @nixpanic as he was already reviewing it

**Review by nixpanic — APPROVED** [review]

**Comment by Yuggupta27** [comment]
> /retest ci/centos/mini-e2e-helm/k8s-1.21

Failed with #2264

**Comment by Yuggupta27** [comment]
> /retest ci/centos/mini-e2e/k8s-1.21

Failed with #2362

**Comment by Yuggupta27** [comment]
> /retest ci/centos/mini-e2e-helm/k8s-1.21

Failed with #2264

**Pull request merged**
